packages/wds-event-tool/src/main/kotlin/org/webdevandsausages/events/dao/EventCRUD.kt

```diff
@@ -74,84 +76,47 @@ class EventCRUD(configuration: Configuration) {
+            val nonNullFields = eventRecord.fields().filter { eventRecord.get(it) != null }
+            eventRecord.store(nonNullFields)
+            findByIdOrLatest(id, transaction)
         }
-
-        return findByIdOrLatest(result.id)
     }
 
-    fun findByParticipantToken(registrationToken: String): Option<EventDto> = db.use { ctx ->
-        val event = Try {
-            ctx.select(*Event.EVENT.fields())
-                .from(Event.EVENT)
-                .leftJoin(Participant.PARTICIPANT)
-                .on(Event.EVENT.ID.eq(Participant.PARTICIPANT.EVENT_ID))
-                .where(Participant.PARTICIPANT.VERIFICATION_TOKEN.eq(registrationToken))
-                .fetchAny()
-                .into(meta.tables.pojos.Event::class.java)
-        }.toOption()
-        when (event) {
-            is Some ->findByIdOrLatest(event.t.id)
-            is None -> event
-        }
-    }
-
-    fun create(event: EventInDto): Option<EventDto> {
-        return with(Event.EVENT) {
-            db.use { ctx ->
-                ctx
-                    .insertInto(
-                        Event.EVENT,
-                        NAME,
-                        SPONSOR,
-                        CONTACT,
-                        DATE,
-                        DETAILS,
-                        LOCATION,
-                        STATUS,
-                        MAX_PARTICIPANTS,
-                        REGISTRATION_OPENS,
-                        VOLUME,
-                        SPONSOR_LINK
-                    )
-                    .values(
-                        event.name,
-                        event.contact,
-                        event.sponsor,
-                        event.date,
-                        event.details,
-                        event.location,
-                        event.status,
-                        event.maxParticipants,
-                        event.registrationOpens,
-                        event.volume,
-                        event.sponsorLink
-                    )
-                    .returning()
-                    .fetchOne()
-            }.let {
-                EventDto(
-                    event = it.into(meta.tables.pojos.Event::class.java)
-                ).toOption()
+    fun findByParticipantToken(registrationToken: String, context: DSLContext = db): Option<EventDto> =
+        context.use { ctx ->
+            val event = Try {
+                ctx.select(*Event.EVENT.fields())
+                    .from(Event.EVENT)
+                    .leftJoin(Participant.PARTICIPANT)
+                    .on(Event.EVENT.ID.eq(Participant.PARTICIPANT.EVENT_ID))
+                    .where(Participant.PARTICIPANT.VERIFICATION_TOKEN.eq(registrationToken))
+                    .fetchAny()
+                    .into(meta.tables.pojos.Event::class.java)
+            }.toOption()
+            when (event) {
+                is Some -> findByIdOrLatest(event.t.id)
+                is None -> event
```

Comment:
Can't remember if I changed in my PR but this passing event to when is kind of awkward. Rather more fluent to just chain with flarMap, something like `.toOption().flatMap { Some(findByIdOrLatest(id)) }`
It will always short circuit if event is None.

Reply:
Nice, this was new for me, I didn't realize it worked that way 👍